ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,168 @@
+from collections import namedtuple
+from datetime import datetime
+
+import ocflib.misc.validators
+from django import forms
+from django.shortcuts import redirect
+from django.shortcuts import render
+from django.urls import reverse
+from jinja2 import Environment
+from jinja2 import PackageLoader
+from ocflib.account.search import user_attrs_ucb
+from ocflib.misc.mail import send_mail
+
+from ocfweb.auth import calnet_required
+from ocfweb.component.forms import wrap_validator
+
+JINJA_MAIL_ENV = Environment(loader=PackageLoader('ocfweb', ''))
+
+
+class NewReservationRequest(namedtuple(
+    'NewReservationRequest', [
+        'real_name',
+        'contact_email',
+        'group',
+        'reason',
+        'date',
+        'starttime',
+        'endtime',
+    ],
+)):
+
+    def to_dict(self):
+        return self._asdict()
+
+
+class RequestForm(forms.Form):
+
+    error_css_class = 'error'
+    required_css_class = 'required'
+
+    real_name = forms.CharField(
+        label='Full Name',
+        widget=forms.TextInput(attrs={'placeholder': 'Oski Bear'}),
+        min_length=3,
+    )
+
+    contact_email = forms.EmailField(
+        label='Contact email',
+        validators=[wrap_validator(ocflib.misc.validators.valid_email)],
+        widget=forms.EmailInput(attrs={'placeholder': 'oski@berkeley.edu'}),
+    )
+
+    verify_contact_email = forms.EmailField(
+        label='Confirm contact email',
+        validators=[wrap_validator(ocflib.misc.validators.valid_email)],
+        widget=forms.EmailInput(attrs={'placeholder': 'oski@berkeley.edu'}),
+    )
+
+    group = forms.CharField(
+        label='Group',
+        widget=forms.TextInput(attrs={'placeholder': 'Open Computing Facility'}),
+        min_length=3,
+    )
+
+    reason = forms.CharField(
+        label='Reason for reservation',
+        widget=forms.Textarea(attrs={'placeholder': ''}),
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (yyyy-mm-dd)',
+        widget=forms.DateInput(attrs={'placeholder': datetime.now().strftime('%Y-%m-%d')}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (24 hour, i.e. 20:00)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (24 hour, i.e. 22:00)',
+    )
```

Comment:
ideally for the dates and times, we'd use a fancy jquery date picker widget or something so that the non-technical masses don't have to figure out how to convert 8pm to 24-hour time, but imo this is plenty good enough to be merged